homeassistant/components/swiss_public_transport/sensor.py

```diff
@@ -121,6 +121,11 @@ def __init__(
     @property
     def native_value(self) -> StateType | datetime:
         """Return the state of the sensor."""
-        return self.entity_description.value_fn(
-            self.coordinator.data[self.entity_description.index]
+        return (
+            self.entity_description.value_fn(
+                self.coordinator.data[self.entity_description.index]
+            )
+            if self.coordinator.data
+            and self.entity_description.index < len(self.coordinator.data)
+            else None
```

Comment:
Why is this changed?

Reply:
was needed for the sensor service, but not anymore for a integration service --> reverted